### DISCUSSION TITLE
[feature]: support for `components/`-templates and `server/api/`-templates

### DISCUSSION THREAD

**Discussion by BananaAcid** [nuxtrdev/nuxtr-vscode#103] [Ideas]
### Environment

--

### What went wrong?

Would be great to have

### How to reproduce it?

--

**Comment by adhamfarrag** [comment]
@BananaAcid I see the value of API templates, but what's the value of a component template ? Can you help me with a use case ?

**Reply by BananaAcid** [reply]
Since we do options api here, all created components need basically the same starter code, which is identical to pages / handling pages. 

https://nuxt.com/docs/guide/directory-structure/components

**Reply by BananaAcid** [reply]
Putting navigation (`Naviagation.vue`) and other vue-blocks in components makes it more managable. Or creating new ones like with attributes and all the extra stuff, just like normal Vue-Components. https://vuejs.org/guide/essentials/component-basics

**Reply by BananaAcid** [reply]
<img width="222" alt="Bildschirmfoto 2023-12-18 um 15 27 07" src="https://github.com/nuxtrdev/nuxtr-vscode/assets/1894723/04dcdc0b-8e72-46be-a930-958268e55ab0">

Example directory structure, this is from a small project. Our normal projects have way more components (usually 20+) and pages (around 10) and api endpoints (like 10) ... larger projects have like 100+ components

**Comment by the94air** [comment]
@adhamfarrag Honestly makes sense to provide a way to apply a custom template for components. My use case is that I want my component templates to not have the style tag by default. I also want my script attributes to have a different order.

**Reply by angelhdzmultimedia** [reply]
```json
"nuxtr.vueFiles.style.addStyleTag": false,
"nuxtr.vueFiles.firstTag": "script | template" 
```

**Reply by angelhdzmultimedia** [reply]
@the94air 👋

**Reply by the94air** [reply]
Thanks for the nudge @angelhdzmultimedia. Works flawlessly. I guess it was more reasonable to set these options at the configuration layer so it can apply to all default templates. Thanks, @adhamfarrag.

**Comment by adhamfarrag** [comment]
Thanks @the94ai, will add it to next release.

**Comment by ghost** [comment]
Hey @adhamfarrag, any update on this? It would be awesome to have this feature

**Comment by kier-mc** [comment]
> Hey @adhamfarrag, any update on this? It would be awesome to have this feature

+1, it would be great to have the customisability of default templates as with layouts/pages, but for components and API routes too.

**Comment by adhamfarrag** [comment]
Just noticed your bing @j4kubj :)  

I'm trying my best to ship this among other features by mid June max. Thanks for binging me about this. 💚

CC: @kier-mc

**Comment by BananaAcid** [comment]
^^ wish we would have this already ... would be useful for the 5 current projects, where I have to tell the folks here over and over again, how the head comments in the components have to look like 😉 ... I am dying inside

**Reply by the94air** [reply]
@BananaAcid you can temporary setup your templates to be ignored in the meantime `-page.template.vue`
https://nuxt.com/docs/api/nuxt-config#ignoreprefix
https://nuxt.com/docs/api/nuxt-config#ignore
